noxfile.py

```diff
@@ -366,12 +363,46 @@ def build_dists(session: nox.Session) -> List[str]:
             "Remove them and try again",
         )
 
+    session.log("# Install build environment")
+    build_venv = Path(session.create_tmp()) / "build-venv"
+    session.run("python", "-m", "venv", "--without-pip", build_venv, silent=True)
+    build_python = build_venv / "bin" / "python"
+    session.run(
+        "python",
+        "-m",
+        "pip",
+        "--python",
+        build_python,
+        "install",
+        "-r",
```

Comment:
Perhaps, document why it's not a constraint?
```suggestion
        "-r",  # can't be constraint dues to regression @ https://github.com/pypa/pip/issues/9243
```